Civi/API/Kernel.php

```diff
@@ -157,17 +157,16 @@ public function runRequest($apiRequest) {
   /**
    * Bootstrap - Load basic dependencies and sanity-check inputs.
    *
-   * @param \Civi\API\V4\Action|array $apiRequest
+   * @param \Civi\Api4\Generic\AbstractAction|array $apiRequest
    * @throws \API_Exception
    */
   public function boot($apiRequest) {
     require_once 'api/Exception.php';
-
-    if (!is_array($apiRequest['params'])) {
-      throw new \API_Exception('Input variable `params` is not an array', 2000);
-    }
     switch ($apiRequest['version']) {
       case 3:
+        if (!is_array($apiRequest['params'])) {
+          throw new \API_Exception('Input variable `params` is not an array', 2000);
+        }
```

Comment:
@eileenmcnaughton this is one performance thing to note: every time you try to access an api `Action` class params as an array it forces the object to serialize its properties. Here it was being done for no reason.

Reply:
Yeah - we no longer need that check IMHO because I think the civicrm_api uses strict type casing